src/utils/getUrlByRequestOptions.ts

```diff
@@ -31,7 +32,9 @@ function getProtocolByRequestOptions(
     return agentProtocol
   }
 
-  return options.cert ? 'https:' : options.uri?.protocol || DEFAULT_PROTOCOL
+  const isSecureRequest = options.cert || options.port === SSL_PORT
+
+  return isSecureRequest ? 'https:' : options.uri?.protocol || DEFAULT_PROTOCOL
```

Comment:
I have only one doubt here. The port could be not the one in the `port` property of `options`.
I you have a look at `getPortByRequestOptions` the port can be also taken from the `agent`. 
I think that we should pass the port to the function.

Reply:
I opted to use the port util in the protocol function as it seemed cleaner than spreading the port with the options?
Obviously not ideal since that util would be used twice, however I found it safer in terms of debugging and with minimal cost associated with it.

```
/**
 * Creates a `URL` instance from a given `RequestOptions` object.
 */
export function getUrlByRequestOptions(options: IsomorphicRequestOptions): URL {
  debug('request options', options)

  const port = getPortByRequestOptions(options)
  // This could mask where the protocol originates making it harder to debug
  const protocol = getProtocolByRequestOptions({ ...options, port })
...
```
What do you think?